### PR TITLE
[Observability] Add additional tags to telemetry data from HTTP client

### PR DESCRIPTION
**Pull request by madushajg**

### CODE REVIEW

File: stdlib/http/src/main/ballerina/src/http/client_endpoint.bal [hunk 1]
```diff
@@ -157,7 +214,14 @@ public type Client client object {
     # + request - An HTTP inbound request message
     # + return - The response for the request or an `http:ClientError` if failed to establish communication with the upstream server
     public remote function forward(@untainted string path, Request request) returns Response|ClientError {
-        return self.httpClient->forward(path, request);
+        Response|ClientError response = self.httpClient->forward(path, request);
+        if (response is Response) {
+            error? x = addObservabilityInformation(path, HTTP_FORWARD, response.statusCode);
```
Comment (nadundesilva) [inline stdlib/http/src/main/ballerina/src/http/client_endpoint.bal]: The HTTP method should be extraced from the request. This is simply a remote function to forward a request

File: stdlib/http/src/main/ballerina/src/http/client_endpoint.bal [hunk 2]
```diff
@@ -204,7 +275,14 @@ public type Client client object {
     # + promise - The related `PushPromise`
     # + return - A promised HTTP `Response` message, or an error if the invocation fails
     public remote function getPromisedResponse(PushPromise promise) returns Response|ClientError {
-        return self.httpClient->getPromisedResponse(promise);
+        Response|ClientError response = self.httpClient->getPromisedResponse(promise);
+        if (response is Response) {
+            error? x = observe:addTagToSpan(HTTP_STATUS_CODE_GROUP, getStatusCodeRange(response.statusCode));
```
Comment (nadundesilva) [inline stdlib/http/src/main/ballerina/src/http/client_endpoint.bal]: We can add the path and method from the push promise can't we ?

File: stdlib/http/src/main/ballerina/src/http/client_endpoint.bal (hunk 1 above)
Comment (madushajg) [reply stdlib/http/src/main/ballerina/src/http/client_endpoint.bal]: Fixed with 527d3559b6c04d5002739eaffbabbfa83ea6856a

File: stdlib/http/src/main/ballerina/src/http/client_endpoint.bal (hunk 2 above)
Comment (madushajg) [reply stdlib/http/src/main/ballerina/src/http/client_endpoint.bal]: Fixed with 527d3559b6c04d5002739eaffbabbfa83ea6856a

**Pull request merged**